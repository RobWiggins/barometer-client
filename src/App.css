* {
  box-sizing: border-box;
  font-family: 'Open Sans', 'Source Sans Pro', 'Lato', sans-serif;
}

html {
  background-color: rgba(29, 161, 242, .2);
  max-width: 1000px;
  min-width: 290px;
}

header {
  padding: 5px;
  font-family: 'Roboto', sans-serif;
}

header > h1 {
  font-size: 45px;
}

main {
  /* height: 90vh; */
  text-align: center;
  margin: auto;
}

body {
  text-align: center;
  margin-left: 15px;
  margin-right: 15px;
  background-color: snow;
  min-height: 100vh;
  border-radius: 1%;
  padding-bottom: 15px;
  min-width: 350px;
  /* what to do here with body background */
}

#searchArea {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: space-between;
  padding-top: 15px;
}

#search-label {
  padding-right: 7px;
}

/* .search > button {
  margin-left: 3px;
  font: white;
  border-radius: 10%;
} */

/* trying icon */
.search img {
  height: 1em;
}

#searchError,
#searchError > p {
  background-color: #ff9800;
  color: white;
  font-weight: 1.5em;
}

.hidden {
  visibility: hidden;
}

.pastSearch {
  padding-left: 10px;
  padding-right: 10px;
  background-color: rgba(221, 221, 221, .5);
  border: none;
  color: black;
  text-align: center;
  font-size: 16px;
  margin: 4px 2px;
  transition: 0.3s;
  cursor: pointer;
  border-radius: 5%;
}

.pastSearch:hover {
  color: white;
  background-color: rgba(29, 161, 242, .6);
}

#searchHistory {
  padding-bottom: 15px;
}

#searchHistory h3 {
  margin-top: 0px;
}

#searchTweets {
  border-radius: 2%;
}

.container{
  position: relative;
  width: 100px;
  height: 100px;
}
.top{
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  width: 100%;
  height: 100%;
  background-color: #0F0;
  opacity: 0;
  transition: opacity 700ms;
}
.loading{
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 20px;
  height: 20px;
  z-index: 0;
}

.loading i{
  animation: rotation 2s infinite linear;
}

@keyframes rotation {
  from {
    transform: rotate(0deg);
  } to {
    transform: rotate(359deg);
  }
}

.chart {
  min-width: 200px;
}

#emotionChart {
  padding: 10px;
}

#listTitle {
  padding-bottom: 10px;
  padding-top: 15px;
  font-size: 25px;
}

#tweets > ul {
  display: flex;
  justify-content: center;
}

#tweets li {
  border-left: solid gray 1px;
  border-right: solid gray 1px;
  border-top: solid gray 1px;
  border-bottom: solid gray 1px;
  border-radius: 1%;
  padding: 10px;
  margin: 7px;
  text-align: left;
}

ul {
  list-style: none;
  padding-left: none;
  padding-inline-start: 0px;
}

@media only screen and (min-width: 768px) {
  
  body {
     
  }

  main {
    /* height: 90vh; */
    text-align: center;
    margin: auto;
  }

  html {
    background-color: rgba(29, 161, 242, .2);
    min-width: 290px;
    max-width: none;
  }

  #grid-holder {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 10px;
    grid-auto-rows: minmax(100px, auto);
    grid-auto-columns: 50%;
  }

  #emotionChart {
    padding: 10px;
    grid-column: 2;
    grid-row: 1;
  }

  #sentimentChart {
    padding-top: 15px;
    grid-column: 2;
    grid-row: 2;
  }

  #tweet-list {
    grid-column: 1;
    grid-row: 1 / 3;
    max-height: 800px;
    overflow: auto;
  }
}
