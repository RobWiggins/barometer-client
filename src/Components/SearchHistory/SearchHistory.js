import React from 'react';

export default class SearchHistory extends React.Component {

  render() {

    let history;
    if (!this.props.queries) {
      history = '';
    } else {
      history = this.props.queries.map( (query) => {
        return <span onClick={e => {
          e.preventDefault();
          console.log(e.target.innerHTML);
          this.props.handleSubmitQuery(e.target.innerHTML);
        }} key={query.id} text={query.query}>{query.query}</span>;
      })
    }

    /* pass in queries from app state*/
    return (
      <section id="searchHistory">
        <h3>Users searching</h3>
        {history}
      </section>
    )
  }

}