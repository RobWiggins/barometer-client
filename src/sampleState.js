const sampleState = {
  watsonEmotionResults: {
    usage: {
      text_units: 1,
      text_characters: 5563,
      features: 2,
    },
    sentiment: {
      document: {
        score: -0.389527,
        label: 'negative',
      },
    },
    language: 'en',
    emotion: {
      targets: [
        {
          text: 'trump',
          emotion: {
            sadness: 0.527292,
            joy: 0.117104,
            fear: 0.107436,
            disgust: 0.172389,
            anger: 0.160808,
          },
        },
      ],
      document: {
        emotion: {
          sadness: 0.22184,
          joy: 0.613288,
          fear: 0.108367,
          disgust: 0.532003,
          anger: 0.565455,
        },
      },
    },
  },
  tweets: [
    "President Trump's proposed executive action for a #2020Census citizenship question will be illegal, unconstitutional, and un-American. Everyone in the country should be pushing back against it. Walk with me briefly as I unpack this... https://t.co/PLaambiNeD",
    'QUESTION FOR @washingtonpost:\n\nTrump⁠—via lawyers⁠—denies a social relationship with Epstein. So why is this info in PARAGRAPH #22:\n\n"[Per his black book], Epstein\'s relationship with Trump was significant...Epstein listed 14 phone numbers for Trump..." https://t.co/c4fZVcYCwL',
    'This staffer said Trump grabbed her by the shoulders and forcibly kissed her at a campaign event. She is now suing Trump for "violating norms of decency."\n\nSomeone else captured this video, just released tonight.\n\nThe instance seems to have been greatly mischaracterized. https://t.co/qAxJjPyrnN',
    'We can’t afford for the media to divide the Democratic Party. @AOC  @SpeakerPelosi please keep your differences on the DL. The media will use any disagreements to further divide and next thing you know Trump is back in office and God help us. #teamAOC #Palante #BX',
    'This staffer said Trump grabbed her by the shoulders and forcibly kissed her at a campaign event. She is now suing Trump for "violating norms of decency."\n\nSomeone else captured this video, just released tonight.\n\nThe instance seems to have been greatly mischaracterized. https://t.co/qAxJjPyrnN',
    'Wait. Trump literally thought that Pete Buttigieg had a Twitter handle based on the sophomoric nickname Trump himself gave him? https://t.co/JNWuGPihmi',
    '*Wins 2019 World Cup, Golden Ball and Golden Boot*\n\nMegan Rapinoe: WE DID IT! I deserve this. \n\nGOP: What happened to humility? So rude and un-American.\n\nNo one:\n\nDonald Trump: I’m so great looking and smart, a true Stable Genius!\n\nGOP: Very good sir, you are wise AND handsome.',
  ],
  queries: [
    { id: 1, query: 'durant', interval: '2019-07-09T18:59:54.676Z' },
    { id: 2, query: 'trump britain', interval: '2019-07-09T18:59:54.676Z' },
    { id: 3, query: 'butterflies', interval: '2019-07-09T18:59:54.676Z' },
    { id: 4, query: 'bumblebee', interval: '2019-07-10T00:04:58.914Z' },
    { id: 17, query: 'earthquake', interval: '2019-07-10T00:19:49.459Z' },
    { id: 18, query: 'trees', interval: '2019-07-10T00:32:34.926Z' },
    { id: 19, query: 'earthquake', interval: '2019-07-10T01:02:39.611Z' },
  ],
  currentQuery: 'trump',
  isSearchDisabled: false, // TODO not used at moment
  hasError: false,
  showLandingPage: false,
  isLoading: false,
};

module.exports = sampleState;
