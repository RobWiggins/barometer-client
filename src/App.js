import React from 'react';
// import { Route, Link } from 'react-router-dom';
import config from './config';
import './App.css';
import Header from './Components/Header/Header.js';
import FormQuery from './Components/FormQuery/FormQuery';
import EmotionChart from './Components/EmotionChart/EmotionChart';
import TweetList from './Components/TweetList/TweetList';
import ExampleChart from './Components/ExampleChart/ExampleChart';
import SentimentChart from './Components/SentimentChart/SentimentChart';
import ExampleSentimentChart from './Components/ExampleSentimentChart/ExampleSentimentChart';
import SearchError from './Components/SearchError/SearchError';
import SearchHistory from './Components/SearchHistory/SearchHistory'

class App extends React.Component {

  state = {
    watsonEmotionResults: null,
    tweets: null,
    queries: null,
    currentQuery: null,
    isSearchDisabled: false, // TODO not used at moment
    hasError: false,
  };

  /* retrieve past query history for all users */
  componentDidMount() {
    fetch(config.API_ENDPOINT + '/queries/history')
    .then(response => {
      if (!response.ok) {
        throw new Error({ message: 'error with getting history'});
      }
      return response.json()
    }).then(data => {
        console.log('QUERY HISTORY ON MOUNT: ', data.queries)
        this.setState( {
          queries: data.queries
        });
    })
    .catch(err => console.log(err.message))
  }
  
  /* Query valid and submitted, add query to history */
  /** TODO im doing this wrong, needs prev props */
  componentDidUpdate() {
    const body = JSON.stringify( {
      query: this.state.currentQuery
    });
    const options = { 
      method: 'POST', 
      headers: { 'content-type': 'application/json' },
      body,
    };

    /** TODO im doing this wrong */
    fetch(config.API_ENDPOINT + '/queries/history', options)
      .then(response => {
        if (!response.ok) {
          throw new Error({ message: 'error with retrieving history'});
        }
        return response.json()
      })
      .then(data => console.log('inserted query to history: ', data))
      .catch(err => console.log(err.message))
  }

  // returns false and disables search function if search > 25ch
  handleSearch(searchQuery) {
    // maximum search query length, allows limiting of watson targets
    //   const queryMaxLength = 25;
    //   if (searchQuery.length > queryMaxLength && !this.state.isSearchDisabled) {
    //     this.setState({
    //       isSearchDisabled: true,
    //   });
    //  } else if (searchQuery.length <= queryMaxLength && this.state.isSearchDisabled) {
    //     this.setState({
    //       isSearchDisabled: false,
    //     });
    //   }
  }

  /* TODO come back and wire up functionally with twitter retrieveTweets() */
  handleSubmitQuery = query => {

    console.log(query);
    /* try to communicate with backend */
    fetch(config.API_ENDPOINT + `/tweets/queries/${query}`) // how to send body?
      .then(response => {
        if (!response.ok) {
          throw new Error({ message: 'something seems to have gone wrong' });
        }
        return response.json();
      })
      .then(data => {
        // console.log('FROM THE FRONT END!!!!!', data);
        console.log('data.watsonEmotionResults: ', data.watsonEmotionResults);
        this.setState({
          watsonEmotionResults: data.watsonEmotionResults,
          tweets: data.duplicatesFiltered,
          hasError: false,
          currentQuery: data.currentQuery
        });
        
      })
      .catch(error =>
        this.setState({
          hasError: true,
        })
      ); // fix error message handling
  };

  render() {
    let isEmotionDataPresent = this.state.watsonEmotionResults ? true : false;
    let isTweetDataPresent = this.state.tweets ? true : false;
    console.log('isEmotionDataPresent: ', isEmotionDataPresent);

    let emotionChartDisplay, sentimentChartDisplay;
    if (isEmotionDataPresent) {
      emotionChartDisplay = (
        <EmotionChart watsonEmotionResults={this.state.watsonEmotionResults} />
      );
      sentimentChartDisplay = (
        <SentimentChart
          watsonEmotionResults={this.state.watsonEmotionResults}
        />
      );
    } else {
      emotionChartDisplay = <ExampleChart />;
      sentimentChartDisplay = <ExampleSentimentChart />;
    }

    let errorDisplay;
    if (this.state.hasError) {
      errorDisplay = <SearchError />;
    } else {
      errorDisplay = '';
    }

    return (
      <main className="main">
        <Header />
        <FormQuery
          isSearchDisabled={this.state.isSearchDisabled}
          handleSearch={this.handleSearch}
          handleSubmitQuery={this.handleSubmitQuery}
        />
        {errorDisplay}
        <SearchHistory queries={this.state.queries}></SearchHistory>
        <div id="grid-holder">
            {emotionChartDisplay}
            {sentimentChartDisplay}
            <TweetList tweets={this.state.tweets} />
        </div>
      </main>
    );
  }
}

export default App;
