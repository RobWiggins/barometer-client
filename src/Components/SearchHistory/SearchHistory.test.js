import React from 'react';
import ReactDOM from 'react-dom';
import SearchHistory from './SearchHistory';
import sampleState from '../../sampleState.js';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../../App'

describe('testing SearchHistory Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchHistory handleSubmitQuery={App.handleSubmitQuery}
      queries={sampleState.queries} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a SearchHistory list with queries and handleSubmit props', () => {
    const wrapper = shallow(<SearchHistory handleSubmitQuery={App.handleSubmitQuery}
      queries={sampleState.queries} />).find('#searchHistory');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});