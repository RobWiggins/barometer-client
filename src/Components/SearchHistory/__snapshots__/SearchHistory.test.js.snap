// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SearchHistory Component renders a SearchHistory list with queries and handleSubmit props 1`] = `
<section
  id="searchHistory"
>
  <h3>
    Users searching
  </h3>
  <ul
    id="buttonsHistory"
  >
    <button
      className="pastSearch"
      key="0"
      onClick={[Function]}
    >
      earthquake
    </button>
    <button
      className="pastSearch"
      key="1"
      onClick={[Function]}
    >
      trees
    </button>
    <button
      className="pastSearch"
      key="2"
      onClick={[Function]}
    >
      earthquake
    </button>
    <button
      className="pastSearch"
      key="3"
      onClick={[Function]}
    >
      bumblebee
    </button>
    <button
      className="pastSearch"
      key="4"
      onClick={[Function]}
    >
      butterflies
    </button>
    <button
      className="pastSearch"
      key="5"
      onClick={[Function]}
    >
      trump britain
    </button>
  </ul>
</section>
`;
